chain/reward.go

```diff
@@ -0,0 +1,30 @@
+package chain
+
+import (
+	"github.com/nknorg/nkn/common"
+	"github.com/nknorg/nkn/util/config"
+)
+
+var (
+	MiningRewards = make(map[uint32]common.Fixed64)
+)
+
+func init() {
+	for i := uint32(0); i < config.TotalRewardDuration; i++ {
+		MiningRewards[i] = common.Fixed64(config.InitialReward - config.ReductionAmount*int(i))
+	}
+}
+
+func GetRewardByHeight(height uint32) common.Fixed64 {
+	if height == 0 {
+		panic("invalid height")
+	}
+	duration := (height - 1) / uint32(config.RewardAdjustInterval)
+	if duration >= config.TotalRewardDuration {
+		return 0
+	}
+
+	reward := MiningRewards[duration]
+	rewardPerBlock := int64(reward) / int64(config.RewardAdjustInterval)
```

Comment:
If we use integer here, only 280M instead of 300M token will be mined.
If we use 1 more digit (0.1 precision), then 298M will be mined.
If we use 2 more digits (0.01 precision), then 299.8 M will be mined.
If we use 3 more digits (0.001 precision), then 299.98M will be mined.